java/ql/lib/semmle/code/java/ControlFlowGraph.qll

```diff
@@ -347,12 +347,28 @@ private module ControlFlowGraphImpl {
     )
   }
 
+  private predicate methodMayThrow(Method m, ThrowableType t) {
+    exists(AstNode n |
+      t = n.(ThrowStmt).getThrownExceptionType() and
+      not n.(ThrowStmt).getParent() = any(Method m0).getBody()
```

Comment:
It this to filter out methods that always throw?

Reply:
Yes.